controllers/drplacementcontrol_controller.go

```diff
@@ -606,22 +606,29 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to get DRPC object")
 	}
 
+	drPolicy, err := r.getDRPolicy(ctx, drpc, logger)
+	if err != nil {
```

Comment:
On errors we would not update DRPC Status which hence can be an issue for users processing the resource as only logs can help determine why the failure occured. We could use recordFailure here, but its ability to handle `nil` Placement object is not clean.